classes/form/settings_form.php

```diff
@@ -161,5 +161,27 @@ public function define_client_selection($mform, $config) {
         $mform->addHelpButton('filesystem', 'settings:clientselection:title', 'tool_objectfs');
         return $mform;
     }
+
+    public function validation($data, $files) {
+        $errors = parent::validation($data, $files);
+
+        if (is_numeric($data['maxtaskruntime']) && $data['maxtaskruntime'] < 0 ) {
```

Comment:
By the way moodle had is_number() function which could be useful if we are trying to validate integers as said in issue description. is_numeric will return true for numbers like 1.3

Reply:
With "duration" form element (PARAM_INT), the value is automated converted to seconds and return as Float number instead of INT. We may convert to INT value if it is strictly required to be integer. Otherwise, I think it is fine to have float value for these elements.

With "Text" form element (PARAM_INT), textbox only accept whole integer number (enter 1.5 - it automatically turn to 1), therefore we do not need to check if they are float or integer.

Reply:
I don't know if it's strictly required to be integer. Please double check it.